### DISCUSSION TITLE
Customize output filename  in the YAML

### DISCUSSION THREAD

**Discussion by juan-dura** [quarto-dev/quarto-cli#9913] [Q&A]
### Description

QUARTO can customize the output filename using quarto_render function. But it would be better to do it in the YAML.
With R Markdown is possible to customize the output filename:
```
title: "Some title"
author: "First Last"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
output: pdf_document
knit: (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(dirname(inputFile), paste0(Sys.Date(),"_Report","_FirstLast",".pdf"))) })
```

**Comment by mcanouil** [comment]
You mean like?

```yaml
format:
  html:
    output-file: my-filename.html
```

See <https://quarto.org/docs/reference/formats/html.html#rendering>.

You cannot do the knitr thing because that is knitr and here you are using Quarto.

**Reply by juan-dura** [reply]
Yes. I know I can use knitr. But it would be nice to be able to change the name of the output file within qmd. For example:
```
format:
  html:
    output-file: paste("my-filename", currentDate(), ".html")
'''

**Reply by cderv** [reply]
> But it would be nice to be able to change the name of the output file within qmd. For example:

Quarto does not support evaluating YAML configuration, especially in a way tied to a computation tool. Quarto aims to be language agnostic and provide great user experience from configuration edition. This would not allow us to provide YAML autocompletion and validation correctly. 

I don't think there will be a way to do this from within the YAML configuration. Using post-render script, or a wrapper around quarto rendering command is easy enough. 

> With R Markdown is possible to customize the output filename:

R Markdown does not offer to change the output file from within the document itself directly. The correct way is to call `rmarkdown::render()` and set the argument `output_file`

This is the same in quarto, at CLI `quarto render --output-file ...` or using the R package to call the CLI with `quarto::quarto_render(..., output_file = ...)`

What you showed is not a R Markdown feature, but a RStudio IDE feature. It is a trick that is used to control the command that RStudio IDE will use to render the document. This YAML you showed in example will only works in RStudio IDE context. Rendering the document in console won't change the filename for you. 

It does not seem they support it for Quarto.

**Reply by juan-dura** [reply]
In that case, it is a pity. Thank you.

**Reply by mcanouil** [reply]
As said, you can use post-render and the environment variable to rename files https://quarto.org/docs/projects/scripts.html#pre-and-post-render
And since you seem to use R, you can use the R wrapper (`quarto`) as suggested by Christophe.

```
knit: (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding = encoding, output_file = file.path(dirname(inputFile), paste0(Sys.Date(),"_Report","_FirstLast",".pdf"))) })
```

As far as I know, this only works in RStudio and not if you render via the R command `rmarkdown::render` so in the end you are in the same spot.

**Reply by cderv** [reply]
> In that case, it is a pity. Thank you.

@juan-dura what I meant is that it would be a feature request for the IDE if you want this same feature to work (i.e. customizing the rendering function of the IDE button using a YAML field)
